cmake/gcc-m-cpu.cmake

```diff
@@ -89,6 +89,8 @@ elseif("${ARCH}" STREQUAL "arc")
     set(GCC_M_CPU em4_fpuda)
   elseif(CONFIG_CPU_HS3X)
     set(GCC_M_CPU archs)
+  elseif(CONFIG_CPU_HS4X)
+    set(GCC_M_CPU hs4x)
```

Comment:
Can you add this to https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/mgmt/mcumgr/grp/os_mgmt/include/os_mgmt_processor.h as a separate commit?

Reply:
Done: https://github.com/zephyrproject-rtos/zephyr/pull/57348